src/reflect/scala/reflect/internal/tpe/GlbLubs.scala

```diff
@@ -184,8 +184,8 @@ private[internal] trait GlbLubs {
   /** Eliminate from list of types all elements which are a supertype
     *  of some other element of the list. */
   private def elimSuper(ts: List[Type]): List[Type] = ts match {
```

Comment:
I don't claim to understand it, but I have a question:
This seems to be an `O(n^2)` operation (i.e. for every element search all remaining). Would it be faster to cache the parents in a `Set` and check the set for inconsistencies for the previously encountered parents instead?
That could make it linear, I think.
(would probably be slower if the number of elements is small)